frontend/src/react-query/api/security.tsx

```diff
@@ -21,44 +22,28 @@ import {
   ListRolesRequestSchema,
   type ListRolesResponse,
   SecurityService,
-} from 'protogen/redpanda/api/console/v1alpha1/security_pb';
-import {
-  type ListRoleMembersRequest as ListRoleMembersRequestDataPlane,
-  ListRoleMembersRequestSchema as ListRoleMembersRequestSchemaDataPlane,
-  type ListRolesRequest as ListRolesRequestDataPlane,
-  ListRolesRequestSchema as ListRolesRequestSchemaDataPlane,
 } from 'protogen/redpanda/api/dataplane/v1/security_pb';
 import { MAX_PAGE_SIZE, type MessageInit, type QueryOptions } from 'react-query/react-query.utils';
 import { useInfiniteQueryWithAllPages } from 'react-query/use-infinite-query-with-all-pages';
 import { formatToastErrorMessageGRPC } from 'utils/toast.utils';
 
 export const useListRolesQuery = (
-  input?: MessageInit<ListRolesRequestDataPlane>,
+  input?: MessageInit<ListRolesRequest>,
   options?: QueryOptions<GenMessage<ListRolesRequest>, ListRolesResponse>,
 ) => {
-  const listRolesRequestDataPlane = create(ListRolesRequestSchemaDataPlane, {
+  const listRolesRequest: ListRolesRequest = create(ListRolesRequestSchema, {
     pageToken: '',
     pageSize: MAX_PAGE_SIZE,
     ...input,
   });
 
-  const listRolesRequest = create(ListRolesRequestSchema, {
-    request: listRolesRequestDataPlane,
-  }) as MessageInit<ListRolesRequest> & Required<Pick<MessageInit<ListRolesRequest>, 'request'>>;
-
   const listRolesResult = useInfiniteQueryWithAllPages(listRoles, listRolesRequest, {
-    pageParamKey: 'request',
     enabled: options?.enabled,
-    getNextPageParam: (lastPage) =>
-      lastPage?.response?.nextPageToken
-        ? {
-            ...listRolesRequestDataPlane,
-            pageToken: lastPage.response?.nextPageToken,
-          }
-        : undefined,
+    getNextPageParam: (lastPage) => lastPage?.nextPageToken || undefined,
+    pageParamKey: 'pageToken',
```

Comment:
@Mateoc Let's make sure this continues to work when there are multiple pages required to fetch, can try it with page size of 1 and check 2 requests are coming through

Reply:
yes it does